### PR TITLE
Adding a new config and plugins for cluter-api-provider-kubevirt

### PR DESCRIPTION
**Pull request by bardielle**

### CODE REVIEW

**Comment by rgolangh** [comment]
@bardielle retitle the PR, its about the cluter-api-provider-kubevirt and not the IPI provider

**Comment by petr-muller** [comment]
/uncc

Please tag me when this PR is ready for review :)

File: ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-master.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    namespace: openshift
+    tag: golang-1.13
+binary_build_commands: make build
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.13
+images:
+- dockerfile_path: Dockerfile.rhel
```
Comment (rgolangh) [inline ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-master.yaml:21]: I sent https://github.com/openshift/cluster-api-provider-kubevirt/pull/3 to rename the file to Dockerfile.rhel

Comment (bardielle) [reply ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-master.yaml:21]: Thanks I approved it

**Comment by bardielle** [comment]
@petr-muller Please review this PR

**Review by chenyosef — CHANGES_REQUESTED** [review]

**Review by chenyosef — APPROVED** [review]

File: ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-master.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+base_images:
+  base:
+    name: "4.6"
```
Comment (petr-muller) [inline ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-master.yaml:3]: Unfortunately there was a branch cut in the meantime, so masters are now 4.7.
```suggestion
    name: "4.7"
```

File: ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-master.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+images:
+- dockerfile_path: Dockerfile.rhel
+  from: base
+  inputs:
+    ocp_builder_rhel-8-golang-openshift-4.6:
+      as:
+      - registry.svc.ci.openshift.org/ocp/builder:rhel-8-golang-openshift-4.6
+  to: kubevirt-machine-controllers
+promotion:
+  name: "4.6"
```
Comment (petr-muller) [inline ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-master.yaml:29]:
```suggestion
  name: "4.7"
```

File: ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-master.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  namespace: ocp
+resources:
+  '*':
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tag_specification:
+  name: "4.6"
```
Comment (petr-muller) [inline ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-master.yaml:37]:
```suggestion
  name: "4.7"
```

File: ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-master.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
+base_images:
+  base:
+    name: "4.6"
+    namespace: ocp
+    tag: base
+  ocp_builder_rhel-8-golang-openshift-4.6:
+    name: builder
+    namespace: ocp
+    tag: rhel-8-golang-openshift-4.6
```
Comment (petr-muller) [inline ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-master.yaml:9]: To prevent confusion -- the `4.6` should stay in this stanza for now.

File: ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-master.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    namespace: openshift
+    tag: golang-1.13
+images:
+- dockerfile_path: Dockerfile.rhel
+  from: base
+  inputs:
+    ocp_builder_rhel-8-golang-openshift-4.6:
+      as:
+      - registry.svc.ci.openshift.org/ocp/builder:rhel-8-golang-openshift-4.6
+  to: kubevirt-machine-controllers
```
Comment (petr-muller) [inline ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-master.yaml:27]: The `4.6` should stay here too (no change needed, just to prevent unnecessary changes)

**Review by petr-muller — COMMENTED** [review]
LGTM, two things:

1. We've done a branch cut to 4.7 in the meantime, so some fields need to go from `4.6` to `4.7` (I've used the suggestions to point them out).
2. You are not setting up automated merges, is that intentional?

**Comment by rgolangh** [comment]
> LGTM, two things:
> 
>     1. We've done a branch cut to 4.7 in the meantime, so some fields need to go from `4.6` to `4.7` (I've used the suggestions to point them out).
> 
>     2. You are not setting up automated merges, is that intentional?

No, we should set it. @bardielle edit core-services/prow/02_config/_config.yaml and duplicate the `openshift/cluster-api-provider-ovirt`

**Review by petr-muller — APPROVED** [review]
